packages/core/src/components/ui/ProductGallery/ProductGalleryPage.tsx

```diff
@@ -4,17 +4,41 @@ import ProductGrid from 'src/components/product/ProductGrid'
 import Sentinel from 'src/sdk/search/Sentinel'
 
 import { ProductCardProps } from 'src/components/product/ProductCard'
-import { useProducts } from './usePageProducts'
+import { memo } from 'react'
+import { usePageProductsQuery } from 'src/sdk/product/usePageProductsQuery'
+import {
+  ProductListingPageContext,
+  SearchPageContext,
+  usePage,
+} from 'src/sdk/overrides/PageProvider'
 
 interface Props {
   page: number
   title: string
   productCard?: Pick<ProductCardProps, 'showDiscountBadge' | 'bordered'>
+  itemsPerPage: number
 }
 
-function GalleryPage({ page, title, productCard }: Props) {
-  const products = useProducts(page) ?? []
-  const { itemsPerPage } = useSearch()
+function ProductGalleryPage({ page, title, productCard, itemsPerPage }: Props) {
+  const {
+    state: { term, sort, selectedFacets },
+  } = useSearch()
+
+  const context = usePage() as ProductListingPageContext | SearchPageContext
+  const productsPerPage = context?.data?.productsPerPage
+
+  const { currentProductsPerPage } = usePageProductsQuery(
```

Comment:
```suggestion
  const { currentPagedProducts } = usePageProductsQuery(
```

Reply:
A variable called `productsPerPage` followed by a `currentProductsPerPage` makes me think they will have the same type or similar values, and it's not the case here, right?

Reply:
hey @gvc , we have changed this strategy while pairing with @icazevedo. as this thread is related to outdated code, I'll resolve then :)